.github/workflows/ghpages.yml

```diff
@@ -0,0 +1,55 @@
+name: GHPages
+
+on:
+  push:
+    branches:
+      - $default_branch
+
+  workflow_dispatch:
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: "Node Cache"
+        uses: actions/cache@v2
+        with:
+          path: ~/.npm
+          key: ${{ runner.os }}-cache-node-${{ hashFiles('**/package.json', '**/package-lock.json') }}
+
+      - name: "Node Cache Modules"
+        uses: actions/cache@v2
+        with:
+          path: node_modules
+          key: ${{ runner.os }}-cache-node_modules-${{ hashFiles('**/package.json', '**/package-lock.json') }}
```

Comment:
As we are caching dependencies with node modules and m2 it makes sense to have at least one generic restore key. If we upgrade one dependency there is no reason to not bring in the cache for the remaining dependencies and let npm and mvn fetch the upgrade dependency.